File: src/Tandem_Bicycle/TandemBicycle.js
function tandemBicycle(redShirtSpeeds, blueShirtSpeeds, fastest) {
    redShirtSpeeds.sort((a, b) => a - b);
    blueShirtSpeeds.sort((a, b) => a - b);
    let totalSpeed = 0;
    for (let i = 0; i < redShirtSpeeds.length; i++) {
        const rider1 = redShirtSpeeds[i];
        const rider2 = blueShirtSpeeds[fastest ? blueShirtSpeeds.length - 1 - i : i];
        totalSpeed += Math.max(rider1, rider2);
    }
    return totalSpeed;
}

const redShirtSpeeds = [5, 5, 3, 9, 2];
const blueShirtSpeeds = [3, 6, 7, 2, 1];

console.log(tandemBicycle(redShirtSpeeds, blueShirtSpeeds, true));

File: src/Tandem_Bicycle/TandemBicycle.java
package Tandem_Bicycle;

import java.util.*;

class TandemBicycle {

  public static int tandemBicycle(int[] redShirtSpeeds, int[] blueShirtSpeeds, boolean fastest) {
    Arrays.sort(redShirtSpeeds);
    Arrays.sort(blueShirtSpeeds);
    int totalSpeed = 0;
    for (int i = 0; i < redShirtSpeeds.length; i++) {
        int rider1 = redShirtSpeeds[i];
        int rider2 = blueShirtSpeeds[fastest ? blueShirtSpeeds.length - 1 - i : i];
      totalSpeed += Math.max(rider1, rider2);
    }
    return totalSpeed;
  }

  public static void main(String[] args) {
    int[] redShirtSpeeds = {5, 5, 3, 9, 2};
    int[] blueShirtSpeeds = {3, 6, 7, 2, 1};
    System.out.println(tandemBicycle(redShirtSpeeds, blueShirtSpeeds, true));
  }
}